src/couch/src/couch_doc.erl

```diff
@@ -228,28 +244,28 @@ validate_docid(Id) ->
     couch_log:debug("Document id is not a string: ~p", [Id]),
     throw({illegal_docid, <<"Document id must be a string">>}).
 
-transfer_fields([], #doc{body=Fields}=Doc) ->
+transfer_fields([], #doc{body=Fields}=Doc, _) ->
```

Comment:
all this is necessary for replication, I think? I'm not sure it's necessary to consider that for the special node-local databases. @davisp what are your thoughts?

Reply:
Nope, even in just a PUT you hit this code through db_doc_req -> from_json_obj_validate -> transfer_fields.